src/Json.Schema.ToDotNet.UnitTests/DataModelGeneratorTests.cs

```diff
@@ -2363,7 +2369,152 @@ public int GetHashCode(C obj)
                 ["D"] = new ExpectedContents()
             };
 
-            Assert.FileContentsMatchExpectedContents(_testFileSystem, expectedContentsDictionary);
+            VerifyGeneratedFileContents(expectedContentsDictionary);
+        }
+
+        [Fact(DisplayName = "DataModelGenerator generates attributes for properties with defaults.")]
+        public void GeneratesAttributesForPropertiesWithDefaults()
+        {
+            const string Schema =
+@"{
+  ""type"": ""object"",
+  ""properties"": {
+    ""intProp"": {
+      ""type"": ""integer"",
+      ""description"": ""An integer property.""
+    },
+    ""intPropWithDefault"": {
+      ""type"": ""integer"",
+      ""description"": ""An integer property with a default value."",
+      ""default"": 42
+    },
+    ""numberProp"": {
+      ""type"": ""number"",
+      ""description"": ""A number property.""
+    },
+    ""numberPropWithDefault"": {
+      ""type"": ""number"",
+      ""description"": ""A number property with a default value."",
+      ""default"": 42.1
+    },
+    ""stringProp"": {
+      ""type"": ""string"",
+      ""description"": ""A string property.""
+    },
+    ""stringPropWithDefault"": {
+      ""type"": ""string"",
+      ""description"": ""A string property with a default value."",
+      ""default"": ""42""
```

Comment:
>42 [](start = 21, length = 2)

you couldn't think up a better string value here? how about "thanks for all the fish"

Reply:
Oooh! I'm going to change that.

---
In reply to: [244508189](https://github.com/Microsoft/jschema/pull/62#discussion_r244508189) [](ancestors = 244508189)